contribution.md

```diff
@@ -0,0 +1,125 @@
+# Contributing
+
+Chipmunk uses [Rust](https://www.rust-lang.org/) for processing the log files while frontend application is built using the [ElectronJS](https://www.electronjs.org/)
+
+## Pre-requisite
+
+To build and run chipmunk on local you will need following languages installed on your system.
+1. Rust
+2. NodeJS
+3. Ruby
+
+To check if you have all the pre-requisite installed or not, chipmunk provides the shell
+script for this purpose. After cloning the repo run following command in your preferred terminal.
+
+```
+sh developing/check.sh
+```
+
+If everything is installed then script should print success messages.
+If all pre-requisite are satisfied, [install](#installing-dependencies) dependencies.
+Follow following steps for installing the missing dependencies.
+
+### Installing Rust
+To install the rust, use the Rust version manager called **rust** which will automatically install
+the Rust on local. Run following command in the terminal.
+
+```
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+Once command finishes processing, check the installed Rust version using
+
+```
+rustc --version
+```
+
+which should print the Rust version on terminal.
+
+### Installing NodeJS
+To install the NodeJS, install NVM(Node Version Manager). Follow [installation](https://github.com/nvm-sh/nvm) guide for installing NVM on local.
+Once NVM in ready on your local, install NodeJS using
+
+```
+nvm install --lts
+```
+
+which will install latest long term NodeJS version. Check if node is installed or not using
+
+```
+node -v
+```
+
+which will print the installed NodeJS version on terminal.
+
+### Installing Ruby
+You must have latest Ruby installed on your system. Prefer your choice of version manager
+either [RBENV](https://github.com/rbenv/rbenv) or [RVM](https://rvm.io/).
+Once version manager is ready to use, install Ruby version `>=2.7.7`.
+
+### Installing dependencies
+This project uses few dependencies from other languages and to install them run
+following command in terminal.
+
+```
+sh developing/install.sh
+```
+
+We are using [yarn](https://yarnpkg.com/) as a package manager. Installing yarn is simple
+as you just have to run following command in the terminal.
+
+```
+npm install -g yarn
+```
+
+## Build
+As mentioned earlier backend is built in Rust which takes care of reading the log files
+and frontend app is built using Electron JS.
```

Comment:
I would add here, something like
### Completed build / rebuild
Completed build/rebuild could be done with 
```
rake developing:clean_rebuild_all
```
We are recommend use this command for the first time of building and also in cases when instance of chipmunk was updated/rebased.